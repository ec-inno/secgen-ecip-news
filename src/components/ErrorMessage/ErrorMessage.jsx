import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import Message from '../Message';

const ErrorMessage = ({ title, error }) => {
  const { t } = useTranslation();
  const [isVisible, setVisibility] = useState(false);

  useEffect(() => {
    if (error.message) {
      setVisibility(true);
    }
  }, [error]);

  return (
    <Message
      variant="error"
      className={isVisible ? '' : 'ecl-u-d-none'}
      onClose={() => setVisibility(false)}
      title={title}
      description={error.message}
      icon={{
        shape: 'notifications--error',
        size: 'l',
      }}
      close={{
        variant: 'ghost',
        label: t('Close'),
        icon: {
          shape: 'ui--close',
          size: 's',
        },
      }}
    />
  );
};

ErrorMessage.propTypes = {
  title: PropTypes.string,
  error: PropTypes.shape({
    message: PropTypes.string,
  }),
};

export default ErrorMessage;
