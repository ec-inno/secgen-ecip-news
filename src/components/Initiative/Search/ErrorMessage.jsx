import React, { useState, useEffect } from 'react';
import { useTranslation } from 'react-i18next';

import Message from '../../Message';

const ErrorMessage = ({ title, error }) => {
  const { t } = useTranslation();
  const [isVisible, setVisibility] = useState(false);

  useEffect(() => {
    if (error.message) {
      setVisibility(true);
    }
  }, [error]);

  return (
    <Message
      variant="error"
      className={isVisible ? '' : 'ecl-u-d-none'}
      onClose={() => setVisibility(false)}
      title={title}
      description={error.message}
      icon={{
        shape: 'notifications--error',
        size: 'l',
      }}
      close={{
        variant: 'ghost',
        label: t('Close'),
        icon: {
          shape: 'ui--close',
          size: 's',
        },
      }}
    />
  );
};

export default ErrorMessage;
