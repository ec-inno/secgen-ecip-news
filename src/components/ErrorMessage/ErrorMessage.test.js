import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from './ErrorMessage';

describe('ErrorMessage', () => {
  it('renders correctly', () => {
    const props = { title: 'title', error: { message: 'foo' } };
    const tree = renderer.create(<ErrorMessage {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
