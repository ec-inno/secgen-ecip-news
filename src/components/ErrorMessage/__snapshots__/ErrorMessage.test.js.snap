// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly 1`] = `
<div
  className="ecl-u-d-none ecl-message ecl-message--error"
  data-ecl-message={true}
  role="alert"
>
  <svg
    aria-hidden="true"
    className="ecl-message__icon ecl-icon ecl-icon--l"
    focusable="false"
  >
    
    
    <use
      xlinkHref="file#notifications--error"
    />
  </svg>
  <div
    className="ecl-message__content"
  >
    <button
      className="ecl-message__close ecl-button ecl-button--ghost"
      data-ecl-message-close={true}
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="ecl-button__container"
      >
        <span
          className="ecl-button__label"
          data-ecl-label={true}
        >
          Close
        </span>
        <svg
          aria-hidden="true"
          className="ecl-button__icon ecl-button__icon--after ecl-icon ecl-icon--s"
          data-ecl-icon={true}
          focusable="false"
        >
          
          
          <use
            xlinkHref="file#ui--close"
          />
        </svg>
      </span>
    </button>
    <div
      className="ecl-message__title"
    >
      title
    </div>
    <p
      className="ecl-message__description"
    >
      foo
    </p>
  </div>
</div>
`;
